core/templates/pages/collection-editor-page/navbar/collection-editor-navbar.directive.spec.ts

```diff
@@ -0,0 +1,505 @@
+      .and.returnValue('collectionId');
+    spyOn(collectionEditorStateService, 'getCollection')
+      .and.returnValue(sampleCollection);
+    spyOn(collectionEditorStateService, 'getCollectionRights')
+      .and.returnValue(
+        new CollectionRights({
+          owner_names: [],
+          collection_id: 1,
+          can_edit: true,
+          can_unpublish: true,
+          is_private: true
+        }));
+    spyOn(undoRedoService, 'onUndoRedoChangeApplied$').and.returnValue(
+      mockUndoRedoChangeEventEmitter);
+    spyOnProperty(collectionEditorStateService, 'onCollectionInitialized')
+      .and.returnValue(mockCollectionInitializedEventEmitter);
+
+    ctrl.$onInit();
+    $rootScope.$apply();
+    mockUndoRedoChangeEventEmitter.emit();
+    $rootScope.$apply();
+    mockCollectionInitializedEventEmitter.emit();
+    $rootScope.$apply();
+
+    expect(ctrl.collection).toBe(sampleCollection);
+  });
+
+  it('should unpublish collection successfully', fakeAsync(function() {
+    ctrl.collection = sampleCollection;
+    ctrl.collectionRights = new CollectionRights({
+      owner_names: [],
+      collection_id: 1,
+      can_edit: true,
+      can_unpublish: true,
+      is_private: false
+    });
+    spyOn(collectionRightsBackendApiService, 'setCollectionPrivateAsync')
+      .and.returnValue(Promise.resolve(null));
+    let collectionRightsSpy = spyOn(
+      collectionEditorStateService, 'setCollectionRights')
+      .and.returnValue(null);
+
+    ctrl.unpublishCollection();
+    tick();
+
+    expect(collectionRightsSpy).toHaveBeenCalled();
+  }));
+
+  it('should not unpublish collection in case of ' +
+    'backend error', fakeAsync(function() {
+    ctrl.collection = sampleCollection;
+    ctrl.collectionRights = sampleCollectionRights;
+    spyOn(collectionRightsBackendApiService, 'setCollectionPrivateAsync')
+      .and.returnValue(Promise.reject(null));
+    let collectionRightsSpy = spyOn(
+      collectionEditorStateService, 'setCollectionRights')
+      .and.returnValue(null);
+
+    ctrl.unpublishCollection();
+    tick();
+
+    expect(collectionRightsSpy).not.toHaveBeenCalled();
+  }));
+
+  it('should open a modal to save changes if ' +
+    'additional meta data needed', fakeAsync(function() {
+    sampleCollection = Collection.create(
+      sampleCollectionBackendObject2);
+    ctrl.collection = sampleCollection;
+    ctrl.collectionRights = new CollectionRights({
+      owner_names: [],
+      collection_id: 1,
+      can_edit: true,
+      can_unpublish: true,
+      is_private: false
+    });
+
+    spyOn($uibModal, 'open').and.returnValue({
+      result: Promise.resolve(['metadata'])
+    });
+    spyOn(urlInterpolationService, 'interpolateUrl')
+      .and.returnValue('url');
+    let saveSpy = spyOn(
+      collectionEditorStateService, 'saveCollection')
+      .and.returnValue(null);
+
+    ctrl.publishCollection();
+    $rootScope.$apply();
+    tick();
+    $rootScope.$apply();
```

Comment:
Do we need to call `$rootScope` twice? If that is the case can you add a comment explaining why?

Reply:
removed.

Reply:
The `$rootScope.$apply();` statments seems to be still there.